### ISSUE TITLE
Why the excessive use of arguments?

### ISSUE THREAD

**Issue by realityking** [GlobalSign/PKI.js#27]
I started digging a bit into PKI.js, I'm not a crypo guy but I like to understand the libraries I'm using.

One thing that makes the code very hard to follow, is that it uses the `argumnets` object somewhat excessively.

For example, take [`org.pkijs.simpl.GENERAL_NAME`](https://github.com/GlobalSign/PKI.js/blob/70d5412b6fec8b1f2bab28bd67fedeb664a0063f/org/pkijs/x509_simpl.js#L171-L193) which has one optional arguments, which could be expressed in the function declaration.

Is there a reason for this or would you accept a (or multiple) pull request(s) to change it?

**Comment by YuryStrozhevsky** [comment]
Youa are talking not about "function" but about "object". Please note that "org.pkijs.simpl.GENERAL_NAME" is a contructor for object. So, you have three options how to work with this object:
1) Create an empty object and then put values directly like "gen_name.Name = test";
2) Initialize the internal values of the object via using "schema". This is useful when you are parsing big object and know that this part of ASN.1 structure must satisfy "GENERAL_NAME" schema. By doing this all internal field of GENERAL_NAME object will be created automatically;
3) Put internal values via "object style": new GENERAL_NAME({ Name: "test", NameType: 1}). It is just for easy processing with object construction;

**Comment by YuryStrozhevsky** [comment]
Hope you were satisfied with my answer.
